### ISSUE TITLE
mockServiceWorker.js is shipped to production

### ISSUE THREAD

**Issue by h-jennings** [mswjs/examples#34]
https://github.com/mswjs/examples/blob/5434b15d9dc22b41faf66745606db561d17411dd/examples/rest-angular/angular.json#L25

Hey there! In the generated `mockServiceWorker.js` file, it states to not ship/serve this file to production. By including this file in the `assets: []` array, angular will package this file in its production build. Is there any way we could tell angular to serve this file for testing and development but _do not_ include it in the production bundle?

Otherwise this is a very helpful example! Thank you for taking the time to create it :)

**Comment by kettanaito** [comment]
Hey, @h-jennings. All the credit for that examples goes to @timdeschryver!

Including the worker script itself is not an issue, as long as you import `msw` and call `worker.start()` conditionally in your app, so it never ends up in the production build. I understand the confusion, but don't know an exact configuration to omit the file during the production build. Some people introduce a custom shell script that removes the `mockServiceWorker.js` from after the production build. If you find out something better please post it here, I'd be glad to update the example, or accept a pull request from you.

**Comment by timdeschryver** [comment]
The example will copy the service worker file into the dist folder, but it won't use it.
I think this doesn't harm anymore, it doesn't get imported, it won't get server neither.

I explored the options within the [angular.json](https://angular.io/guide/workspace-config) and according to this [Stack Overflow question](https://stackoverflow.com/questions/53297722/exclude-assets-in-production-build) you should be able to do it. For me, this seems a lot of work and is error-prone because you will have to manage the assets for multiple environments, especially for an example.


This reminds me I wanted to add a note in the "where is my public directory" section in the [docs](https://mswjs.io/docs/getting-started/integrate/browser#where-is-my-public-directory). I kinda need some opinions on this because we can go two ways.

It could either use the current setup, or it could use a second setup (which I discovered recently).

You can also just add the service worker to the assets folder (this is the public directory for an Angular application). This means you wouldn't need to add it angular build process, but it would mean you would have to point MSW to the current URL. Personally, I think the first option (how it currently is) is the best.


```ts
	worker.start({
		serviceWorker: {
			url: "/assets/mockServiceWorker.js",
		},
	});
```

**Comment by h-jennings** [comment]
@timdeschryver Thank you so much for the thoughtful reply and helpful suggestions!

I ended up stumbling upon the same solution you mentioned, where you create a separate config (I named it 'dev' in my case) and pass it along to `ng serve --configuration=serve-dev`. I definitely agree that in a large project, having to manually manage these configurations is far from ideal. I'll run it by my team and see which they would prefer, managing two environments or just allowing the code to be built in production (like you mentioned, shouldn't be an issue since it's never referenced)

Thank you again!
